### PR TITLE
Introduce `pdf.ocr.correct`.

### PR DESCRIPTION
**Pull request by koaning**
This PR adds a new `pdf.ocr.correct` recipe meant to allow the selected spans from `pdf.image.manual` to be ocr'ed by tessaract. There are [limits](https://explosion-ai.slack.com/archives/C0241LQED4G/p1696857710446579?thread_ts=1696857109.870059&cid=C0241LQED4G) to this approach, mainly explained here but as-is it's not an unreasonable feature.

### CODE REVIEW

File: prodigy_pdf/__init__.py [hunk 1]
```diff
@@ -86,3 +88,109 @@ def before_db(examples):
...
+    dataset: str,
+    source: str,
+    labels: str,
+    scale: int = 3,
+    remove_base64:bool=False,
+    fold_dashes:bool = False,
+    autofocus: bool = False
+) -> ControllerComponentsDict:
+    """Applies OCR to annotated segments and gives a textbox for corrections."""
+    stream = get_stream(source)
```
Comment (magdaaniol) [inline prodigy_pdf/__init__.py:152]: You're not importing `get_stream`, are you? Didn't it break for you? Weird

File: prodigy_pdf/__init__.py [hunk 2]
```diff
@@ -86,3 +88,109 @@ def before_db(examples):
...
+    labels = labels.split(",")
```
Comment (magdaaniol) [inline prodigy_pdf/__init__.py]: Nit, but we have a type mismatch. Labels are first str and then they become a list.

File: prodigy_pdf/__init__.py [hunk 3]
```diff
@@ -86,3 +88,109 @@ def before_db(examples):
...
+
+    def new_stream(stream):
+        for ex in stream:
+            useful_spans = [span for span in ex['spans'] if span['label'] in labels]
```
Comment (magdaaniol) [inline prodigy_pdf/__init__.py]: If there's an example that doesn't happen to have any `spans` e.g. a page in a PDF with no annotations at all, that would throw a KeyError

Comment (koaning) [reply prodigy_pdf/__init__.py]: Good catch. Replacing with `.get()`.

**Pull request merged**